packages/playback-core/src/index.ts

```diff
@@ -23,174 +30,41 @@ export const generatePlayerInitTime = () => {
-  autoPlay: boolean | ValueOf<AutoplayTypes>;
-  autoplay: boolean | ValueOf<AutoplayTypes>;
-};
-
-export type HTMLMediaElementProps = Partial<Pick<HTMLMediaElement, 'src'>>;
-
-export type MuxMediaProps = HTMLMediaElementProps & MuxMediaPropTypes;
-export type MuxMediaPropsInternal = MuxMediaProps & {
-  playerSoftwareName: Options['data']['player_software_name'];
-  playerSoftwareVersion: Options['data']['player_software_version'];
-};
-
-export const toPlaybackIdParts = (playbackIdWithOptionalParams: string): [string, string?] => {
-  const qIndex = playbackIdWithOptionalParams.indexOf('?');
-  if (qIndex < 0) return [playbackIdWithOptionalParams];
-  const idPart = playbackIdWithOptionalParams.slice(0, qIndex);
-  const queryPart = playbackIdWithOptionalParams.slice(qIndex);
-  return [idPart, queryPart];
-};
-
 export const toMuxVideoURL = (playbackId?: string, { domain = MUX_VIDEO_DOMAIN } = {}) => {
   if (!playbackId) return undefined;
   const [idPart, queryPart = ''] = toPlaybackIdParts(playbackId);
   return `https://stream.${domain}/${idPart}.m3u8${queryPart}`;
 };
 
-export const inferMimeTypeFromURL = (url: string) => {
-  let pathname = '';
-  try {
-    pathname = new URL(url).pathname;
-  } catch (e) {
-    console.error('invalid url');
-  }
-
-  const extDelimIdx = pathname.lastIndexOf('.');
-  if (extDelimIdx < 0) return '';
-
-  const ext = pathname.slice(extDelimIdx + 1);
-  const upperExt = ext.toUpperCase();
-
-  return isKeyOf(upperExt, ExtensionMimeTypeMap) ? ExtensionMimeTypeMap[upperExt] : '';
-};
-
-export const getType = (props: Partial<Pick<MuxMediaProps, 'type' | 'src'>>) => {
-  const type = props.type;
-
-  if (type) {
-    const upperType = type.toUpperCase();
-
-    return isKeyOf(upperType, MimeTypeShorthandMap) ? MimeTypeShorthandMap[upperType] : type;
-  }
-
-  const { src } = props;
-
-  if (!src) return '';
-
-  return inferMimeTypeFromURL(src);
+export const getError = (mediaEl: HTMLMediaElement) => {
+  return muxMediaState.get(mediaEl)?.error;
 };
 
-export const getStreamTypeConfig = (streamType?: ValueOf<StreamTypes>) => {
-  // for regular live videos, set backBufferLength to 8
-  if ([StreamTypes.LIVE, StreamTypes.DVR].includes(streamType as any)) {
-    const liveConfig = {
-      backBufferLength: 8,
-    };
-
-    return liveConfig;
-  }
-
-  // for LL Live videos, set backBufferLenght to 4 and maxFragLookUpTolerance to 0.001
-  if ([StreamTypes.LL_LIVE, StreamTypes.LL_DVR].includes(streamType as any)) {
-    const liveConfig = {
-      backBufferLength: 4,
-      maxFragLookUpTolerance: 0.001,
-    };
-
-    return liveConfig;
-  }
+export const initialize = (
+  props: Partial<MuxMediaPropsInternal>,
+  mediaEl?: HTMLMediaElement | null,
+  core?: PlaybackCore
+) => {
+  // Automatically tear down previously initialized mux data & hls instance if it exists.
+  teardown(mediaEl, core);
 
-  return {};
-};
+  muxMediaState.set(mediaEl as HTMLMediaElement, {});
```

Comment:
idea(non-blocking): Since we've moved to a factory model, we can likely refactor the module-level state to be defined here in the closure, correct?

Reply:
yes, would love to get this done as a fast follow if possible